provisioning/device/src/transport_interface.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { X509 } from 'azure-iot-common';
+
+export namespace DeviceProvisioningTransport {
+  /**
+   * @private
+   * Callback for responses from the previsioning service
+   */
+  export type ResponseCallback = (err?: Error, result?: any) => void;
```

Comment:
>result?: an [](start = 47, length = 11)

responses in our node packages typically have a third parameter that contains the raw response we get back from the service.  Utilized for debug purposes.

Reply:
true only for HTTP though, wouldn't surface this in the case of a transport-agnostic thing.

---
In reply to: [142788003](https://github.com/Azure/azure-iot-sdk-node/pull/100#discussion_r142788003) [](ancestors = 142788003)